cfgov/teachers_digital_platform/models/activity_index_page.py

```diff
@@ -0,0 +1,528 @@
+        return final_facets
+
+    # TODO: Remove this function when we switch to DSL
+    def get_nested_facets(self, class_object, narrowed_facets, selected_facets, parent=None):  # noqa: E501
+        if not parent:
+            flat_final_facets = [
+                {
+                    'selected': result['id'] in selected_facets,
+                    'id': result['id'],
+                    'title': result['title'],
+                    'parent': result['parent'],
+                } for result in class_object.objects.filter(pk__in=narrowed_facets).get_ancestors(True).values('id', 'title', 'parent')]  # noqa: E501
+            final_facets = []
+            root_facets = [root_facet for root_facet in flat_final_facets if root_facet['parent'] is None]  # noqa: E501
+            for root_facet in root_facets:
+                children_list = self.get_nested_facets(class_object, narrowed_facets, selected_facets, root_facet['id'])  # noqa: E501
+                child_selected = any(
+                    child['selected'] is True or child['child_selected'] is True for child in children_list  # noqa: E501
+                )
+                final_facets.append(
+                    {
+                        'selected': root_facet['selected'],
+                        'child_selected': child_selected,
+                        'id': root_facet['id'],
+                        'title': root_facet['title'],
+                        'parent': root_facet['parent'],
+                        'children': children_list
+                    })
+            return final_facets
+        else:
+            children = [
+                {
+                    'selected': result['id'] in selected_facets or result['parent'] in selected_facets,  # noqa: E501
+                    'id': result['id'],
+                    'title': result['title'],
+                    'parent': result['parent'],
+                    'children': self.get_nested_facets(class_object, narrowed_facets, selected_facets, result['id']),  # noqa: E501
+                    'child_selected': any(
+                        child['selected'] is True or child['child_selected'] is True for child in  # noqa: E501
+                        self.get_nested_facets(class_object, narrowed_facets, selected_facets, result['id'])  # noqa: E501
+                    )
+                } for result in class_object.objects.filter(pk__in=narrowed_facets).filter(parent_id=parent).values('id', 'title', 'parent')]  # noqa: E501
+            return children
+
+    class Meta:
+        verbose_name = "TDP Activity search page"
+
+
+def default_nested_facets(class_object):
+    """Build a nested facet tree, initially only for ActivityTopic objects."""
+    setup = []
+    parents = class_object.objects.filter(parent=None)
+    default_attrs = {
+        "selected": False,
+        "child_selected": False,
```

Comment:
@higs4281 It looks like "child_selected" will always be set to False?

Reply:
> child_selected" will always be set to False?

The approach here is to start with a cached facet tree that matches the initial state of the activity index page: No search terms, no facet selections, no queries required.

Then, rather than building a new facet structure every time with Elasticsearch queries, we can adjust the cached default facet tree to match a request.